controller/handler/dashboard.js

```diff
@@ -15,41 +15,83 @@ const database = require('../../model');
  * @returns {View} Rendered page
  */
 function dashboardView (request, reply) {
-    const trial = database.sequelize.model('trial');
+    const todayDate = moment();
 
-    trial
-    .findAll()
-    .then((trials) => {
-        // Process data into format expected in view
-        const trialData = trials.map(processTrial);
+    return Promise
+        .all([
+            database.sequelize.query(
+                `
+                SELECT *
+                FROM trial AS tr
+                LEFT JOIN (
+                	SELECT st.trialId, COUNT(pa.id) AS recruitedCount
+                	FROM stage AS st
+                	JOIN patient AS pa
+                	ON pa.stageId = st.id
+                	GROUP BY st.trialId
+                ) AS recruited
+                ON tr.id = recruited.trialId
+                LEFT JOIN (
+                	SELECT st.trialId, COUNT(pa.id) AS completedCount
+                	FROM stage AS st
+                	JOIN patient AS pa
+                	ON pa.stageId = st.id
+                	WHERE pa.dateCompleted > ?
+                	GROUP BY st.trialId
+                ) AS completed
+                ON tr.id = completed.trialId
+                LEFT JOIN (
+                	SELECT st.trialId, COUNT(pa.id) AS activeCount
+                	FROM stage AS st
+                	JOIN patient AS pa
+                	ON pa.stageId = st.id
+                	WHERE ? BETWEEN pa.dateStarted AND pa.dateCompleted
+                	GROUP BY st.trialId
+                ) AS active
+                ON tr.id = active.trialId;
+                `,
+                {
+                    type: database.sequelize.QueryTypes.SELECT,
+                    replacements: [
+                        todayDate.toISOString(),
+                        todayDate.toISOString()
+                    ]
+                }
+            )
+        ])
+        .then((data) => {
+            const trials = data[0];
```

Comment:
@ChristianMurphy : In future, if I use multiple queries, term trials would be too specific.